src/node_snapshotable.cc

```diff
@@ -91,6 +92,17 @@ const SnapshotData* NodeMainInstance::GetEmbeddedSnapshotData() {
   return ss.str();
 }
 
+const std::vector<intptr_t>& SnapshotBuilder::CollectExternalReferences() {
+  static auto registry = std::make_unique<ExternalReferenceRegistry>();
+  return registry_->external_references();
+}
```

Comment:
```suggestion
auto SnapshotBuilder::registry_ = std::make_unique<ExternalReferenceRegistry>();

const std::vector<intptr_t>& SnapshotBuilder::CollectExternalReferences() {
  return registry_->external_references();
}
```
I think this should fix the compilation error.

Reply:
```suggestion
std::unique_ptr<ExternalReferenceRegistry> registry = nullptr;

const std::vector<intptr_t>& SnapshotBuilder::CollectExternalReferences() {
  static auto registry = std::make_unique<ExternalReferenceRegistry>();
  if (!registry_) {
    registry_ = std::move(registry);
  }
  return registry_->external_references();
}
```
Or this if you wanna initialize it only in the first call.

Reply:
```suggestion
std::unique_ptr<ExternalReferenceRegistry> SnapshotBuilder::registry_ = nullptr;

const std::vector<intptr_t>& SnapshotBuilder::CollectExternalReferences() {
  static auto registry = std::make_unique<ExternalReferenceRegistry>();
  if (!registry_) {
    registry_ = std::move(registry);
  }
  return registry_->external_references();
}
```
Or this if you wanna initialize it only in the first call.